### ISSUE TITLE
Always show 404 page on all pages except root

### ISSUE THREAD

**Issue by zoxon** [vite-pwa/astro#15]
All links no longer work 
After clicking on any link if developer console closed I see a 404 page, if open noting happens

My config
```js
AstroPWA({
  mode: 'development',
  base: '/',
  scope: '/',
  registerType: 'autoUpdate',
  manifest,
  includeAssets: ['**/favicons/*.{png,svg,ico,xml}'],
  workbox: {
    navigateFallback: '/404',
    globPatterns: ['**/*.{css,js,html,svg,png,ico,txt}'],
  },
  devOptions: {
    enabled: true,
    navigateFallbackAllowlist: [/^\/404$/],
  },
}),
```
   
![image](https://user-images.githubusercontent.com/3866196/228197714-fd30a198-6599-4fa5-a853-cfd0330e1556.png)

**Comment by userquin** [comment]
@zoxon weird, can you provide a minimal repro? I'll check it later...

**Comment by userquin** [comment]
@zoxon tested with lates Astro version and it is working properly

**Comment by zoxon** [comment]
@userquin I tried to create a minimal reproduction, but without success. Everything works as expected, not sure what the problem is.

Do you have any ideas or ways of debugging?

**Comment by zoxon** [comment]
If change config to `navigateFallback: null` and remove `html` from `globPatterns`

I get the error "Uncaught (in promise) non-precached-url: createHandlerBoundToURL('/') was called, but that URL is not precached. Please pass in a URL that is precached instead." and stay on the same page. 

But it works when developer console is closed

![image](https://user-images.githubusercontent.com/3866196/228767492-59b25e70-4c02-4657-ade9-f06282089d59.png)

**Comment by userquin** [comment]
@zoxon can you check https://github.com/vite-pwa/astro/issues/16#issuecomment-1531434408 ?

**Comment by Clarkkkk** [comment]
I encounter the similar problem. Here is the [reproduction](https://github.com/Clarkkkk/vite-pwa-astro-reproduction).

`npm run dev` seems ok.

But in `npm run build` and `npm run preview`, all routes are point to the root html.

It seems in Astro the `navigateFallback` should not be set , which is for single page app according to the [workbox doc](https://developer.chrome.com/docs/workbox/reference/workbox-build/).

I patch the package to disable the `navigateFallback`, and the issue is gone.
```js
// src/index.ts
if (useWorkbox.navigateFallback === undefined)
      useWorkbox.navigateFallback = config.base ?? config.vite?.base ?? '/'
```

```js
// astro.config.ts
// ...
workbox: {
  // ...
  navigateFallback: null
}
```

**Comment by userquin** [comment]
@Clarkkkk if you set the fallback to null, offline will not work

**Comment by userquin** [comment]
@Clarkkkk it seems a problem with Astro 3, the pwa integration hook being called before Astro builds the dist folder (check there is a workbox-build warning about regex):

![imagen](https://github.com/vite-pwa/astro/assets/6311119/8a80a8fb-96f8-410a-93c6-50e0332fa530)

**Comment by Clarkkkk** [comment]
@userquin I notice that too. But it turns out that the plugin will be called twice, before and after the generation of html files. And the generated `sw.js` contains the correct `precacheAndRoute`.

**Comment by userquin** [comment]
@Clarkkkk I don't think so, first [build] at the bottom should be before the workbox warning:

![imagen](https://github.com/vite-pwa/astro/assets/6311119/9da05697-e9b0-4c94-8b45-d3daf10fbdd9)

**Comment by userquin** [comment]
check also the sitemap, there is no entry with last modified, only the paths

**Comment by userquin** [comment]
your Base.astro cannot resolve /src/pwa.ts, changing the src to relative:

![imagen](https://github.com/vite-pwa/astro/assets/6311119/f03f3536-c194-4f5b-8a9d-d9f0d32d1116)

**Comment by Clarkkkk** [comment]
> @Clarkkkk I don't think so, first [build] at the bottom should be before the workbox warning:
> 
> ![imagen](https://user-images.githubusercontent.com/6311119/275349048-9da05697-e9b0-4c94-8b45-d3daf10fbdd9.png)

I am not familiar with the whole process. It is called twice in my test, for example using something like `globPatterns: ["**/*.{js,css}", "**/*.impossible"]`, and it outputs with two warnings.

I notice that you're using `globPatterns:  "{**/*.{svg,jpg,js,css,html}"`, which should be `globPatterns:  "**/*.{svg,jpg,js,css,html}"`, is that relavant?

**Comment by Clarkkkk** [comment]
> check also the sitemap, there is no entry with last modified, only the paths

Thank you, I simply use a start template, didn't check that.



> your Base.astro cannot resolve /src/pwa.ts, changing the src to relative:
> 
> ![imagen](https://user-images.githubusercontent.com/6311119/275349368-f03f3536-c194-4f5b-8a9d-d9f0d32d1116.png)

I copy those lines from the [doc](https://vite-pwa-org.netlify.app/frameworks/astro.html#auto-update), and it works on my windows, but relative path is better.

**Comment by userquin** [comment]
If you add N entries to the glob patterns array, you will get N logs, 1 per entry. Astro generates the dist folder after the astro:build:done hook or in paralel (first [buid] log at bottom after the warnings).
